sources/mongodb/helpers.py

```diff
@@ -106,6 +107,41 @@ def _filter_op(self) -> Dict[str, Any]:
                 filt[self.cursor_field]["$gt"] = self.incremental.end_value
 
         return filt
+    
+    def _projection_op(self, projection) -> Optional[Dict[str, Any]]:
+        """Build a projection operator.
+        
+        A tuple of fields to include or a dict specifying fields to include or exclude.
+        The incremental `primary_key` needs to be handle differently for inclusion
+        and exclusion projections.
+
+        Returns:
+            Tuple[str, ...] | Dict[str, Any]: A tuple or dictionary with the projection operator.
+        """
+        if projection is None:
+            return None
+
+        projection_dict = dict(_fields_list_to_dict(projection, "projection"))
+
+        # NOTE we can still filter on primary_key if it's excluded from projection
```

Comment:
Mongo has "_id" column which is a kind of primary key. Currently when we create a resource for a collection, we set the primary_key to "_id". if that field is removed, dlt will later complain that primary key is not assuming any value so I'd add it back with a warning (like you do below). that we can do way earlier (when we create a resource)

In very rare cases when someone sets the primary key to something else (on resource / incremental) your code blow is fine. Adding back primary key and warning is good.

cool you thought about this edge case!

and in 99% of cases this is what you'll find in the incremental. Now the question is if